Syntax error in CICE_RunMod.F90

There is a syntax error on line 840 of https://github.com/ACCESS-NRI/cice4/blob/access-esm1.5/drivers/access/CICE_RunMod.F90 on the `access-esm1.5` branch. The line should say
```
      real (kind=dbl_kind) :: cszn ! counter for history averaging
```

### Details

Running
```
spack install cice4@access-esm1.5%intel@2021.8.0 arch=linux-rocky8-x86_64
``` 
results in 
```
...
     286    mpif90 -c -r8 -i4 -O2 -free -align all -w -ftz -convert big_endian -assume byterecl -no-vec -xHost -fp-model precise  -I. -I/scratch/tm70/pcl851/tmp/spack-stage/spack-stage-cice4-access-esm1.5-kqnj7ol67pkfdqxq5pxefvz3hdjqgk
            6g/spack-src/drivers/access -I/scratch/tm70/pcl851/tmp/spack-stage/spack-stage-cice4-access-esm1.5-kqnj7ol67pkfdqxq5pxefvz3hdjqgk6g/spack-src/source -I/scratch/tm70/pcl851/tmp/spack-stage/spack-stage-cice4-access-esm1.5-kqn
            j7ol67pkfdqxq5pxefvz3hdjqgk6g/spack-src/mpi -I/scratch/tm70/pcl851/tmp/spack-stage/spack-stage-cice4-access-esm1.5-kqnj7ol67pkfdqxq5pxefvz3hdjqgk6g/spack-src/csm_share -I. -I/g/data/tm70/pcl851/src/ACCESS-NRI/release/linux-
            rocky8-x86_64/intel-2021.8.0/oasis3-mct-access-esm1.5-4ozzazkbvdaaqjxb3zqrrhzpza2dsk4f/include/psmile.MPI1 -I/g/data/tm70/pcl851/src/ACCESS-NRI/release/linux-rocky8-x86_64/intel-2021.8.0/oasis3-mct-access-esm1.5-4ozzazkbvda
            aqjxb3zqrrhzpza2dsk4f/include -I/g/data/tm70/pcl851/src/ACCESS-NRI/release/linux-rocky8-x86_64/intel-2021.8.0/netcdf-fortran-4.6.1-c7pn3evewkepbdtkkkhjnt5p6hcaykdz/include CICE_RunMod.f90
  >> 287    CICE_RunMod.f90(840): error #5082: Syntax error, found '&' when expecting one of: %FILL <IDENTIFIER>
     288          real (kind=dbl_kind) :: & cszn
     289    ------------------------------^
     290    compilation aborted for CICE_RunMod.f90 (code 1)
  >> 291    make: *** [/scratch/tm70/pcl851/tmp/spack-stage/spack-stage-cice4-access-esm1.5-kqnj7ol67pkfdqxq5pxefvz3hdjqgk6g/spack-src/bld/Makefile:121: CICE_RunMod.o] Error 1
     292    exit 2
...
```

--- a/drivers/access/CICE_RunMod.F90
+++ b/drivers/access/CICE_RunMod.F90
@@ -896,7 +896,7 @@
          i,j         , & ! horizontal indices
          ilo,ihi,jlo,jhi ! beginning and end of physical domain
 
-      real (kind=dbl_kind) :: & cszn
+      real (kind=dbl_kind) :: cszn ! counter for history averaging
 
       call ice_timer_start(timer_column)
 

